src/components/pages/EnrollmentRoute.tsx

```diff
@@ -0,0 +1,64 @@
+import { compact } from 'lodash-es';
+import { Navigate } from 'react-router-dom';
+
+import Loading from '../elements/Loading';
+
+import NotFound from './404';
+
+import useSafeParams from '@/hooks/useSafeParams';
+import { useHasClientPermissions } from '@/modules/permissions/useHasPermissionsHooks';
+import generateSafePath from '@/utils/generateSafePath';
+
+const EnrollmentsRoute: React.FC<
+  React.PropsWithChildren<{
+    clientIdParam?: string;
+    enrollmentIdParam?: string;
+    view?: boolean;
+    edit?: boolean;
+    redirectRoute?: string;
+    getRouteParams?: (params: {
+      clientId?: string;
+      enrollmentId?: string;
+    }) => object;
+  }>
+> = ({
+  clientIdParam = 'clientId',
+  enrollmentIdParam = 'enrollmentId',
+  view = false,
+  edit = false,
+  redirectRoute,
+  getRouteParams = (x) => x,
+  children,
+}) => {
+  const { [clientIdParam]: clientId, [enrollmentIdParam]: enrollmentId } =
+    useSafeParams();
+
+  const [allowed, { loading, data }] = useHasClientPermissions(
+    clientId || '',
+    compact([
+      view ? 'canViewEnrollmentDetails' : null,
+      edit ? 'canEditEnrollments' : null,
+    ])
```

Comment:
You can't edit things you can't view, as a rule.
We should be able to remove the `view` prop, from both this and `ClientRoute`. And replace this with:
```suggestion
     [edit ? 'canEditEnrollments' : 'canViewEnrollmentDetails'],
```

Reply:
Fixed